client/src/lib/utils/Application/InsertApplication.ts

```diff
@@ -0,0 +1,96 @@
+import { SupabaseClient, PostgrestError } from "@supabase/supabase-js";
+import { Database } from "../../types/database.types";
+import { createOpportunity } from "@/lib/utils/Application/createOpportunity";
+import { Profile } from "../../store/profile";
+import { Application } from "@/lib/store/track";
+
+export async function InsertApplication(
+  supabase: SupabaseClient,
+  profile: Profile,
+  application: Application
+): Promise<string | PostgrestError | undefined> {
+  const { data: opportunityData, error: opportunityError } = await supabase
+    .from("opportunity")
+    .select("opportunity_id")
+    .eq("role_title", application.role_title)
+    .eq("company_name", application.company_name)
+    .eq("type", application.type);
+
+  let opportunityId;
+  if (opportunityError) {
+    console.log("Error finding opportunity:", opportunityError.message);
+  }
+
+  if (opportunityData && opportunityData.length > 0) {
+    console.log("Opportunity found:", opportunityData[0].opportunity_id);
+    opportunityId = opportunityData[0].opportunity_id;
+  } else {
+    opportunityId = await createOpportunity(
+      supabase,
+      application.role_title,
+      application.company_name,
+      application.type as Database["public"]["Enums"]["OpportunityType"]
+    );
+  }
+
+  if (!opportunityId) {
+    console.error("Failed to find or create opportunity");
+    return;
+  }
+
+  const newApplication: Omit<
```

Comment:
Why omit?

Reply:
application_id is already and will be created through supabase column "application_id" so that is why I put omit to disregard the application_in generated on user side.